R/symptome_compartment_class.R

```diff
@@ -0,0 +1,226 @@
+#' An SEIR model for compartments of different symptoms
+#'
+#' Note:
+#' 1) Sum of poulation equals to 1 at any time point.
+#' An error will be reported the sum of initial population is not equal to 1.
+#' 2) At this stage, we only consider disease-caused death cases
+#' calculating mortality.
+#'
+#' model structure:
+#'    ---------------------------(omega)---------------------------------
+#'    |                                                                 |
+#'    |                                                                 |
+#'    --> S --(lam)--> E --(gamma) ------(1-ihr)-------> I ---(nu.i)--> R <--
+#'                                 |                     |                  |
+#'                                 |                 (pdeath.i)             |
+#'                                 |                     |                  |
+#'                                 |            --> CM <--                  |
+#'                                 |            |                           |
+#'                                 |      (pdeath.hsa)                      |
+#'                                 |            |                           |
+#'                                 -- (ihr)---> HA --(nu.hsa)----------------
+
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Class definitions
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+#' @describeIn model_symptome to set up dependent classes for model_symptome
+setClass("ggplot")
+setClass("gg")
+
+#' An S4 class of an basic SEIR model with compartments of different symptomes
+#'
+#' @slot name A string gives the name of the model
+#' @slot initial_population A list of the initial population size of each group
+#' @slot parameters A list of the model parameters
+#' @slot output A dataframe holding the ode simulation output, with columns as population groups and rows as simulation time points
+#' @slot plot_output A ggplot holding the plot of the ode simulation output
+#'
+#' @import deSolve
+#' @import ggplot2
+#' @import reshape2
+#' @import plyr
+#' @import magrittr
+
+setClass(Class = "model_symptome",
```

Comment:
I'm not sure about this name for the class: I'd go for something more indicative like SEIaIsR (if that's the type of model it is, that is?).